### DISCUSSION TITLE
How do I inject a dependency to a file/directory without explicitly mentioning it in the README

### DISCUSSION THREAD

**Discussion by mfts** [pkgxdev/pkgx#328] [Help]
I need `python3` to run `prepare-commit-msg` (notably without a file extension) but it's a python file (see https://github.com/mfts/tea-demos/blob/feat/ai-commit-msg/ai-commit-msg/prepare-commit-msg.py). 
This file gets placed in `.git/hooks` directory of my current directory and executes on `git commit`. 

I don't want to change the user's behaviour of forcing the user to do something like 
```sh
tea +python.org^3 +git-scm.org git commit
```
But I want python in the environment whenever `git commit` runs in that directory.

Something else I tried: changing the shebang in Line 1 on prepare-commit-msg to `#!/usr/bin/env -S tea python3`
unfortunately, that results in this error message
```
env: tea: No such file or directory
```

**Comment by mxcl** [comment]
The shebang will work.

```
env: tea: No such file or directory
```

Means that `tea` is not in your `PATH` as far as `git` is concerned. How are you doing your commits?

**Reply by mxcl** [reply]
If you are just doing `git commit` and `tea --version` works then it is weird that the environment your `git` calls its hooks from doesn’t use `PATH`
